codec-http2/src/main/java/io/netty/handler/codec/http2/Http2RemoteFlowController.java

```diff
@@ -52,12 +52,26 @@
         int size();
 
         /**
-         * Signal an error and release any retained buffers.
+         * Called to indicate that an error occurred before this object could be completely written.
+         * <p>
+         * The {@link Http2RemoteFlowController} will make exactly one call to either {@link #error(Throwable)} or
+         * {@link #writeComplete()}.
+         * </p>
+         *
          * @param cause of the error.
          */
         void error(Throwable cause);
 
         /**
+         * Called after this object has been successfully written.
+         * <p>
+         * The {@link Http2RemoteFlowController} will make exactly one call to either {@link #error(Throwable)} or
+         * {@link #writeComplete()}.
+         * </p>
+         */
+        void writeComplete();
```

Comment:
We also need to update the docs for `write` to indicate that it should not change the stream state and if it throws an exception `error` will be called immediately after.